src/addrman.cpp

```diff
@@ -1074,6 +1118,15 @@ size_t AddrManImpl::size() const
     return vRandom.size();
 }
 
+size_t AddrManImpl::size(std::optional<Network> net, std::optional<bool> in_new) const
+{
+    LOCK(cs);
+    Check();
+    auto ret = size_(net, in_new);
+    Check();
```

Comment:
It's not needed, but this pattern is also used for other const methods (`GetAddr()`, `Select()`), plus we have a few mutables in AddrMan - since `Check()` is only run if the user enables it, I guess it's better to invoke it once too often rather than not?